plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1627,13 +1637,36 @@ public ConnectorTableHandle beginStatisticsCollection(ConnectorSession session,
     @Override
     public void finishStatisticsCollection(ConnectorSession session, ConnectorTableHandle tableHandle, Collection<ComputedStatistics> computedStatistics)
     {
-        UpdateProperties updateProperties = transaction.updateProperties();
-        Map<String, Integer> columnNameToId = transaction.table().schema().columns().stream()
+        IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
+        Table table = transaction.table();
+        if (handle.getSnapshotId().isEmpty()) {
+            // No snapshot, table is empty
+            verify(
+                    computedStatistics.isEmpty(),
+                    "Unexpected computed statistics that cannot be attached to a snapshot because none exists: %s",
+                    computedStatistics);
+
+            // TODO (https://github.com/trinodb/trino/issues/15397): remove support for Trino-specific statistics properties
+            // Drop all stats. Empty table needs none
+            UpdateProperties updateProperties = transaction.updateProperties();
+            table.properties().keySet().stream()
+                    .filter(key -> key.startsWith(TRINO_STATS_PREFIX))
+                    .forEach(updateProperties::remove);
+            updateProperties.commit();
+
+            transaction.commitTransaction();
+            transaction = null;
+        }
+        long snapshotId = handle.getSnapshotId().orElseThrow();
+
+        Map<String, Integer> columnNameToId = table.schema().columns().stream()
                 .collect(toImmutableMap(nestedField -> nestedField.name().toLowerCase(ENGLISH), Types.NestedField::fieldId));
         Set<Integer> columnIds = ImmutableSet.copyOf(columnNameToId.values());
 
+        // TODO (https://github.com/trinodb/trino/issues/15397): remove support for Trino-specific statistics properties
         // Drop stats for obsolete columns
-        transaction.table().properties().keySet().stream()
+        UpdateProperties updateProperties = transaction.updateProperties();
```

Comment:
can we have a `EnvSinglenodeCompatibility` product test where we make sure that the old stats table properties are actually getting deleted?

Reply:
I work on M1, so my product test creation capabilities are currently severely limited.

Reply:
discussed offline and decided not to add these tests